paz/models/classification/cnn2Plus1.py

```diff
@@ -0,0 +1,354 @@
+    return x
+
+
+def CNN2Plus1D(weights=None, input_shape=(38, 96, 96, 3), seed=305865,
+               architecture: Architecture_Options = 'CNN2Plus1D',
+               tmp_weights_path=None):
+    """Binary Classification for videos with 2+1D CNNs.
+    # Arguments
+        weights: ``None`` or string with pre-trained dataset. Valid datasets
+            include only ``VVAD-LRS3``.
+        input_shape: List of integers. Input shape to the model in following format: (frames, height, width, channels)
+        e.g. (38, 96, 96, 3).
+        seed: Integer. Seed for random number generator.
+        architecture: String. Name of the architecture to use. Currently supported: 'CNN2Plus1D', 'CNN2Plus1D_Filters',
+            'CNN2Plus1D_Layers', 'CNN2Plus1D_Light'. 'CNN2Plus1D_18' is only available without weights.
+
+    # Reference
+        - [A Closer Look at Spatiotemporal Convolutions for Action Recognition](https://arxiv.org/abs/1711.11248v3)
+        - [Video classification with a 3D convolutional neural network]
+        (https://www.tensorflow.org/tutorials/video/video_classification#load_and_preprocess_video_data)
+    """
+    if len(input_shape) != 4:
+        raise ValueError(
+            '`input_shape` must be a tuple of 4 integers. '
+            'Received: %s' % (input_shape,))
+    options = get_args(Architecture_Options)
+    assert architecture in options, f"'{architecture}' is not in {options}"
+
+    random.seed(seed)
+    initializer_glorot_output = tf.keras.initializers.GlorotUniform(seed=random.randint(0, 1000000))
+
+    # input_shape = (None, 10, HEIGHT, WIDTH, 3)
+    image = Input(shape=input_shape, name='image')
+    x = image
+
+    if architecture == 'CNN2Plus1D':
+        x = normal(x, input_shape[1], input_shape[2])
+        if tmp_weights_path is None:
+            tmp_weights_path = "../../../../CLUSTER_OUTPUTS/CNN2Plus1D/2023_10_05-22_08_31/cnn-2plus1d_weights-21.hdf5"
+    elif architecture == 'CNN2Plus1D_Filters':
+        x = filters(x, input_shape[1], input_shape[2])
+        if tmp_weights_path is None:
+            tmp_weights_path = ("../../../../CLUSTER_OUTPUTS/CNN2Plus1DFilters/2023_10_20-14_11_13/" +
+                                "cnn-2plus1d-filters_weights-21.hdf5")
+    elif architecture == 'CNN2Plus1D_Layers':
+        x = layers(x, input_shape[1], input_shape[2])
+        if tmp_weights_path is None:
+            tmp_weights_path = ("../../../../CLUSTER_OUTPUTS/CNN2Plus1DLayers/2023_10_16-10_14_48/" +
+                                "cnn-2plus1d-layers_weights-17.hdf5")
+    elif architecture == 'CNN2Plus1D_Light':
+        x = light(x, input_shape[1], input_shape[2])
+        if tmp_weights_path is None:
+            tmp_weights_path = ("../../../../CLUSTER_OUTPUTS/CNN2Plus1DLight/2023_10_10-11_26_58/" +
+                                "cnn-2plus1d-light_weights-35.hdf5")
+    elif architecture == 'CNN2Plus1D_18':
+        x = original_18(x, input_shape[1], input_shape[2])
```

Comment:
All weights are now uploaded and you could change their path for automatic weight download.